t/secrets.t

```diff
@@ -245,6 +245,29 @@ EOF
 	have_secret "$v/auto-generated-certs-a/ca:certificate";
 	my $ca = secret "$v/auto-generated-certs-a/ca:certificate";
 
+	sub get_cert_validity {
+		use Time::Piece;
+		my ($info) = @_;
+		my $pattern = "%b%n%d %H:%M:%S %Y";
+		my @i = $info =~ qr/Not Before:\s(.*\s+\d{4})\s+([^\n\r]*)\s+Not After\s+:\s(.*\s+\d{4})\s+([^\n\r]*)/m;
+		return undef unless $i[1] eq $i[3]; # ensure timezones are the same
+		return (Time::Piece->strptime($i[2], $pattern) - Time::Piece->strptime($i[0], $pattern));
+	}
+
+	# Check correct TTL
+	my $fixed_ca = qx(safe get $v/fixed/ca:certificate | openssl x509 -inform pem -text);
+	is get_cert_validity($fixed_ca), (5*365*24*3600), "CA cert has a 5 year validity period";
```

Comment:
Have you thought about using `safe x509 show` instead of relying on OpenSSL x509 utils?

```
$ safe x509 show secret/ssl/ca
secret/ssl/ca:
  cn=ca.ssl

  expires in 3350 days
  valid from Dec 8 2017 - Dec 6 2027 (~10 years)

  for the following purposes:
    (no special key usage constraints present)

  for the following names:
    - ca.ssl (DNS)
```

The `expires in` should match your TTL, without relying on Time::Piece.

Reply:
I used the openssl because it is the base truth, not a derived fact, and all the existing tests use openssl.  I can probably switch to `safe x509 show` but not sure its worth it as this code already does what is needed.

Reply:
Time::Piece is core, so no CPAN module needed.

Reply:
We should change the existing tests as well.